src/include/gurt/telemetry_common.h

```diff
@@ -155,6 +155,9 @@ enum {
 	D_TM_SERVER_PROCESS		= 0x000,
 	D_TM_SERIALIZATION		= 0x001,
 	D_TM_RETAIN_SHMEM		= 0x002,
+	D_TM_RETAIN_SHMEM_IF_NON_EMPTY	= 0x004,
```

Comment:
oh, this tag is only for deleting the job metrics by default.  so if D_TM_RETAIN_SHMEM is set, then job metrics will be kept even after the job exit.   Otherwise if D_TM_RETAIN_SHMEM_IF_NON_EMPTY is set, then job metrics will be delete if no threads/processes left.  Though once these share memory is attached to agent, there maybe more options here.